spec/ParseUser.spec.js

```diff
@@ -14,6 +14,27 @@ const Config = require('../lib/Config');
 const cryptoUtils = require('../lib/cryptoUtils');
 
 describe('Parse.User testing', () => {
+  it('retrieves original object when user logs in with third party auth', async () => {
+    let counter = 0;
+    Parse.Cloud.afterSave(Parse.User, async request => {
+      if (!request.original) {
+        const user = request.object;
+        expect(user.get('authData')).not.toBeUndefined();
```

Comment:
If the afterSave trigger is not executed, then the "expects" inside won't apply. You could use a var outside the trigger and move the "expects" outside, or expect that the trigger has been called.

For example:

```
let user;

Parse.Cloud.afterSave(...) {
  user = request.object;
}

expect(user).toBeDefined();
...
```

Reply:
afterSave is supposed to be triggered when logged in with Facebook to update the access token, but I changed the code as you requested.

Reply:
Can you remove the remaining expect out of the trigger and remove the counter variable, I don't think it's necessary?

Reply:
Could you simplify it to this?

```
it('retrieves original object when user logs in with third party auth', async () => {
     const provider = getMockFacebookProvider();
     Parse.User._registerAuthenticationProvider(provider);
     await Parse.User.logInWith('facebook');
     await Parse.User.logOut();

     let original;
     let user;
     Parse.Cloud.afterSave(Parse.User, async request => {
       original = request.original;
       user = request.object;
     });
     await Parse.User.logInWith('facebook');
     expect(original).toBeDefined();
     expect(user.get('authData')).not.toBeUndefined();
   });
```

Reply:
Done, I just put afterSave before because if you put as you suggested it would be triggered only once, so it has to be before so it gets triggered twice as intended (first should be for signup and second for login).